Modules/socketmodule.c

```diff
@@ -7342,38 +7342,34 @@ PyInit__socket(void)
     if (m == NULL)
         return NULL;
 
-    Py_INCREF(PyExc_OSError);
-    PyModule_AddObject(m, "error", PyExc_OSError);
+    if (PyModule_AddObjectRef(m, "error", PyExc_OSError) != 0)
+        return NULL;
     socket_herror = PyErr_NewException("socket.herror",
                                        PyExc_OSError, NULL);
     if (socket_herror == NULL)
         return NULL;
-    Py_INCREF(socket_herror);
-    PyModule_AddObject(m, "herror", socket_herror);
+    if (PyModule_AddObjectRef(m, "herror", socket_herror) != 0)
+        return NULL;
     socket_gaierror = PyErr_NewException("socket.gaierror", PyExc_OSError,
-        NULL);
+                                         NULL);
     if (socket_gaierror == NULL)
         return NULL;
-    Py_INCREF(socket_gaierror);
-    PyModule_AddObject(m, "gaierror", socket_gaierror);
-    PyModule_AddObjectRef(m, "timeout", PyExc_TimeoutError);
+    if (PyModule_AddObjectRef(m, "gaierror", socket_gaierror) != 0)
+        return NULL;
+    if (PyModule_AddObjectRef(m, "timeout", PyExc_TimeoutError) != 0)
```

Comment:
Seems like this wasn't INCREFed before? Surprised that wouldn't cause issues.

Reply:
`timeout` used `PyModule_AddObjectRef` previously as well, so incref wasn't needed and isn't needed now.

The change to `timeout` is to check the return value (I did this here but not elsewhere to match local convention, since we were already checking return value in this function for other things added to the module)

Reply:
Ah thanks, I missed the `Ref` in the old code.